geomstats/learning/knn.py

```diff
@@ -0,0 +1,93 @@
+"""The KNN classifier on manifolds."""
+
+from sklearn.neighbors import KNeighborsClassifier
+
+from geomstats.geometry.euclidean import EuclideanMetric
+
+
+class KNearestNeighborsClassifier(KNeighborsClassifier):
+    """Classifier implementing the k-nearest neighbors vote on manifolds..
+
+    Parameters
+    ----------
+    n_neighbors : int, optional (default = 5)
+        Number of neighbors to use by default.
+    weights : str or callable, optional (default = 'uniform')
+        weight function used in prediction.  Possible values:
+        - 'uniform' : uniform weights.  All points in each neighborhood
+          are weighted equally.
+        - 'distance' : weight points by the inverse of their distance.
+          in this case, closer neighbors of a query point will have a
+          greater influence than neighbors which are further away.
+        - [callable] : a user-defined function which accepts an
+          array of distances, and returns an array of the same shape
+          containing the weights.
+    metric : callable or string, default EuclideanMetric
```

Comment:
See my next comment a few lines later: this argument should be an object of the class RiemannianMetric, for consistency with KMeans and TangentPCA.

Reply:
The docstring needs to be updated.

But now I see why you wanted the class and not the object: in order to be flexible with respect to the dimension?

In this case, I would recommend:
- `metric=None` in the `__init__` signature, as a default value,
- `if metric is None:` condition in the `__init__`, that can access the data's dimension (from metric' params? or something else?) and create the Euclidean metric of the corresponding dimension?

Reply:
What version do you prefer? Your current code, or this proposal in my comment above?